cluster-autoscaler/utils/pod/pod.go

```diff
@@ -79,3 +82,13 @@ func ClearPodNodeNames(pods []*apiv1.Pod) []*apiv1.Pod {
 	}
 	return newPods
 }
+
+func PodRequests(pod *apiv1.Pod) apiv1.ResourceList {
+	inPlacePodVerticalScalingEnabled := utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling)
+	podLevelResourcesEnabled := utilfeature.DefaultFeatureGate.Enabled(features.PodLevelResources)
+
+	return resourcehelper.PodRequests(pod, resourcehelper.PodResourcesOptions{
```

Comment:
@tallclair this helper expects opts determined by the caller, which means Cluster Autoscaler will need to know of any new feature gates related to Pod requests calculation. Should the helper read the feature gates on its own instead so that the callers (like Cluster Autoscaler) don't need to care?

Reply:
I'm also not worried that CA administrators might not be used to passing feature gates to the CA component at all (I know we don't in GKE). CA does accept the standard `--feature-gates` flag, but to my knowledge it has been used very sparingly so far (CA uses internal flags for its own features, and I don't think we've had many k8s experimental features affecting CA like this one).

Do we have a place in k8s docs where we list which components should a given feature flag be passed to? I tried to find one but without any luck.

Reply:
Right, it's also a problem. However it only affects overrides from the default, i.e. when IPPR becomes enabled by default in Beta then CA will recognize it without any flags passing. It should not block us from submitting I think.

Reply:
I don't think CA needs to know the feature gates to use this. I think the two flags that are relevant to feature gates are:
- `UseStatusResources`: this decides whether to factor in the resources from `pod.status.containerStatuses[*].resources`. The function to evaluate those is to take the maximum, so if those aren't set it will be a no-op. The main reason this option exists is that there are cases where you want to compute the requests from _only_ the desired resources (independent of feature gate).
- `SkipPodLevelResources`: This one is a little trickier, but if pod level resources are set, I think they should be used. The apiserver will drop pod level resources if the feature gate isn't enabled in the control plane.

Reply:
OK so we don't have to worry about feature gates not passed to CA, that's good. My worry is still about any flags that may be added in the future that we may miss, from future KEPs. For example `UseStatusResources` if not set would default to false, which means not taking into account the new feature. We knew about IPPR work and we're setting this flag to true, but it may not be the case in general.